src/main/java/com/checkmarx/ast/wrapper/CxWrapper.java

```diff
@@ -216,6 +217,21 @@ public List<Project> projectList(String filter) throws IOException, InterruptedE
         return Execution.executeCommand(withConfigArguments(arguments), logger, Project::listFromLine);
     }
 
+    public ScanResult ScanVorpal(String fileSource, boolean vorpalLatestVersion) throws IOException, InterruptedException, CxException {
+        this.logger.info("Fetching Vorpal scanResult");
+
+        List<String> arguments = new ArrayList<>();
+        arguments.add(CxConstants.CMD_SCAN);
+        arguments.add(CxConstants.SUB_CMD_VORPAL);
+        arguments.add(CxConstants.FILE_SOURCE);
+        arguments.add(fileSource);
+        if (vorpalLatestVersion) {
+            arguments.add(CxConstants.VORPAL_LATEST_VERSION);
+        }
+
+        return Execution.executeCommand(withConfigArguments(arguments), logger, ScanResult::fromLine);
```

Comment:
if you have error, how it handles it? error in the CLI or error that not part of the scan results object error

Reply:
In case of an error, the java-wrapper's CxException will be thrown. added test:
void testScanVorpal_WhenMissingFileExtension_ReturnFileExtensionIsRequiredFailure() throws Exception {
        // ScanResult scanResult = wrapper.ScanVorpal("src/test/resources/python-vul-file", false);
        Assertions.assertThrowsExactly(CxException.class, () -> {
            wrapper.ScanVorpal("src/test/resources/python-vul-file", false);
        }, "file must have an extension");
    }

Reply:
We dont want to throw an error and break the ID.. We want to show the errors in the IDE and the wrapper should return the error message